bson/src/main/org/bson/Document.java

```diff
@@ -154,9 +154,8 @@ public Document append(final String key, final Object value) {
     @SuppressWarnings("unchecked")
     public <T> T get(final Object key, final T defaultValue) {
         notNull("defaultValue", defaultValue);
-        Class<T> clazz = notNull("clazz", (Class<T>) defaultValue.getClass());
         Object value = documentAsMap.get(key);
-        return value == null ? defaultValue : clazz.cast(value);
+        return value == null ? defaultValue : (T) value;
```

Comment:
Note that after type erasure, this is effectively a no-op to keep the compiler happy, since T has no bounds and gets erased to Object.